lib/yargs-factory.ts

```diff
@@ -2054,7 +2060,6 @@ export class YargsInstance {
       this.#isGlobalContext = false;
 
       const handlerKeys = this.#command.getCommands();
-      const requestCompletions = this.#completion!.completionKey in argv;
       const skipRecommendation = helpOptSet || requestCompletions || helpOnly;
```

Comment:
Since `--get-yargs-completions` is _removed_ from `argv` when `strip-dashed` is set to true I changed this check in two ways:

1. do it the same loop the built-in commands `--help` and `--version` are checked
2. check both possible keys (`--get-yargs-completions` and `--getYargsCompletions`)

I did _not_ have to change the other use of `this.#completion.completionKey` elsewhere in this file as it is used to inspect `args` (not cleaned up), not `argv` (which has been cleaned up).

Reply:
The possibility of user-specified parser configurations like `skip-dashed` makes looking for options in `argv` rather painful! I did a local fix in #2308 for one case, but I think we would benefit from a utility routine to make it convenient and robust to find options in `argv`. There are multiple places where it is potentially an issue, like here, for the minority of people who are changing the parser configuration.

Reply:
Do you think I should adjust the method for finding the option? I was thinking it was safe as is here for this specific option as the user should never see it since it'll short circuit when set. Maybe if they made their own option with the same name it would cause trouble, but I think it already would anyway? I haven't verified.

Reply:
I don't think you need to fix it here, but I would want to apply the (unwritten) high-level fix here too.